apps/transport/test/unlock/batch_metrics_test.exs

```diff
@@ -0,0 +1,51 @@
+defmodule Unlock.BatchMetricsTest do
+  use ExUnit.Case, async: true
+  import DB.Factory
+
+  doctest Unlock.BatchMetrics, import: true
+
+  setup do
+    :ok = Ecto.Adapters.SQL.Sandbox.checkout(DB.Repo)
+    pid = Process.whereis(Unlock.BatchMetrics)
+    Ecto.Adapters.SQL.Sandbox.allow(DB.Repo, self(), pid)
```

Comment:
```suggestion
    # because the upserts happen in a separate process (the GenServer),
    # we must tweak the sandbox to be able to see them from here    
    pid = Process.whereis(Unlock.BatchMetrics)
    Ecto.Adapters.SQL.Sandbox.allow(DB.Repo, self(), pid)
```